t/plugin/dynamic-upstream.t

```diff
@@ -202,3 +202,155 @@ GET /server_port?name=1234
 1980
 --- no_error_log
 [error]
+
+
+
+=== TEST 4: add operation of `~~` in vars
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                ngx.HTTP_PUT,
+                [[{
+                    "uri": "/server_port",
+                    "plugins": {
+                        "dynamic-upstream": {
+                            "rules": [
+                                {
+                                    "match": [
+                                        {
+                                            "vars": [
+                                                [ "arg_name","~~","[a-z]+" ]                                            
+                                            ]
+                                        }                            
+                                    ],
+                                    "upstreams": [
+                                        {
+                                           "upstream": {
+                                                "name": "upstream_A",
+                                                "type": "roundrobin",
+                                                "nodes": {
+                                                   "127.0.0.1:1981":20
+                                                }
+                                            },
+                                            "weight": 2
+                                        }
+                                    ]
+                                }
+                            ]
+                        }
+                    },
+                    "upstream": {
+                            "type": "roundrobin",
+                            "nodes": {
+                                "127.0.0.1:1980": 1
+                            }
+                    }                    
+                }]]
+                )
+            ngx.status = code
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 5: regular check failed, and return 1980 port.
+--- request
+GET /server_port?name=1234
+--- response_body eval
+1980
+--- no_error_log
+[error]
+
+
+
+=== TEST 6: allow match configuration to be empty
```

Comment:
need a test case to confirm which upstream will be hit

Reply:
confirm the empty `match` can work fine

Reply:
Test case added.